src/main/java/org/wso2/carbon/connector/amazonsqs/utils/Utils.java

```diff
@@ -0,0 +1,283 @@
+                resultElement = AXIOMUtil.
+                        stringToOM("<" + elementName
+                                + "></" + elementName + ">");
+            }
+        } catch (XMLStreamException | OMException e) {
+            log.error("Error while generating OMElement from element name" + elementName, e);
+        }
+        return resultElement;
+    }
+
+    public static String createUrl(MessageContext messageContext, SqsConnection sqsConnection)
+            throws MalformedURLException, SqsInvalidConfigurationException {
+        try {
+            String queueName = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_NAME);
+            String queueId = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_ID);
+            String queueUrl = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_URL);
+            if (StringUtils.isEmpty(queueName) && (StringUtils.isEmpty(queueId) || StringUtils.isEmpty(queueUrl))) {
+                throw new SqsInvalidConfigurationException("Missing queue info: Should provide value of queue url or " +
+                        "queue name and queue id.");
+            }
+            if (StringUtils.isNotEmpty(queueUrl)) {
+                return queueUrl;
+            }
+            return new URL(Constants.HTTPS, "sqs".concat(sqsConnection.getConnectionConfig().getRegion()).
+                    concat(".amazonaws.com"), Constants.SLASH.concat(queueId).
+                    concat(Constants.SLASH).concat(queueName)).toString();
+        } catch (MalformedURLException e) {
+            throw new MalformedURLException(e.getMessage());
+        }
+    }
+
+    public static void add200ResponseWithOutBody(SqsResponseMetadata responseMetadata,
+                                                 MessageContext messageContext, String parentElementName) {
+        OMElement resultElement = Utils.createOMElement(parentElementName, null);
+        createResponseMetaDataElement(responseMetadata, messageContext, resultElement);
+        setStatusCode(messageContext, "200");
+    }
+
+    public static void createResponseMetaDataElement(SqsResponseMetadata responseMetadata,
+                                                     MessageContext messageContext, OMElement resultElement) {
+        resultElement.addChild(Utils.createOMElement("ResponseMetadata", responseMetadata.requestId()));
+        SOAPBody soapBody = messageContext.getEnvelope().getBody();
+        JsonUtil.removeJsonPayload(((Axis2MessageContext) messageContext).getAxis2MessageContext());
+        ((Axis2MessageContext) messageContext).getAxis2MessageContext().
+                removeProperty(PassThroughConstants.NO_ENTITY_BODY);
+        soapBody.addChild(resultElement);
+    }
+
+    public static void addErrorResponse(MessageContext messageContext, AwsServiceException e) {
+        OMElement resultElement = Utils.createOMElement("ErrorResponse", null);
```

Comment:
Does the `AwsServiceException` contain all the following details for every error? If not this may throw NPE

Reply:
Yes, Reference: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/handling-exceptions.html#sdkserviceexception-and-subclasses